### ISSUE TITLE
[Bug]: Issue parsing PPTs with Images in Background

### ISSUE THREAD

**Issue by Jash271** [run-llama/llama_index#18847]
### Bug Description

I have a use-case to parse ppts, all of which have images in the background 
However the code snippet provided below fails to parse the ppt, with the same issue 
Here's a sample ppt 

[Presentation1.pptx](https://github.com/user-attachments/files/20433930/Presentation1.pptx)

### Version

 0.12.37

### Steps to Reproduce

```
from llama_index.readers.file import PptxReader
from llama_index.core import SimpleDirectoryReader

parser = PptxReader()
file_extractor = {
    ".pptx": parser,
}
documents  = SimpleDirectoryReader(
    input_files=["Presentation1.pptx"],file_extractor=file_extractor
).load_data()

print(documents)


```
OR 

`
docs = SimpleDirectoryReader(input_files=["Presentation1.pptx"]).load_data()
`

Both these code snippets fail to parse the ppt 

### Relevant Logs/Tracbacks

```shell
Failed to load file Presentation1.pptx with error: Make sure that a `_reorder_cache` function is correctly implemented in transformers.models.gpt2.modeling_gpt2 to enable beam search for <class 'transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel'>. Skipping...
```

**Comment by junsiknss** [comment]
Same issue. Does anyone solve this?

**Comment by mirallm** [comment]
@dosubot i tried to upgrade it still same error

Lib\site-packages\transformers\models\vit\feature_extraction_vit.py:30: FutureWarning: The class ViTFeatureExtractor is deprecated and will be removed in version 5 of Transformers. Please use ViTImageProcessor instead.
  warnings.warn(
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
                                                                                                                                                                            We strongly recommend passing in an `attention_mask` since your input_ids may be padded. See https://huggingface.co/docs/transformers/troubleshooting#incorrect-output-when-padding-tokens-arent-masked.
You may ignore this warning if your `pad_token_id` (50256) is identical to the `bos_token_id` (50256), `eos_token_id` (50256), or the `sep_token_id` (None), and your input is not padded.
                                                                                                                                                                            Failed to load file C:\d\temp_files\01S3EXYTJQTKP7ANI57ZH3U5PLODLDWOD5_8c725df1-6752-4d98-a49e-322576d70f76.pptx with error: Make sure that a `_reorder_cache` function is correctly implemented in transformers.models.gpt2.modeling_gpt2 to enable beam search for <class 'transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel'>. Skipping...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:15<00:00, 15.72s/it]

**Comment by mirallm** [comment]
you need to install pip install transforms <4.5

**Comment by mirallm** [comment]
@dosubot but its very slow, i cant parse like 5 at the same time

**Comment by Jash271** [comment]
Hey @AstraBert  @logan-markewich 

I've noticed gaps in the existing pptx reader, like not being able to parse charts, tables or read slide notes 
Given that parsing images has been causing an issue, I am thinking to make parsing images optional ( as it may or may not always add value to slide content ) and adding a flag called -> explain using llm 
To consolidate all the data in the slide and provide an explanation of the slide intent and key points, so that we have enriched context that can be used for retrieval ? 

Mind if I assign this issue to myself ?

**Comment by Miarije** [comment]
I have the same issue despite all mentioned packages being up to date, it breaks due to captioning images that I do not need captioned. @Jash271 I would love your solution of making it optional!

**Comment by logan-markewich** [comment]
Feel free to make a PR anyone, ill review it

**Comment by Jash271** [comment]
Alright yeah! I an working on it, will raise a PR soon

**Comment by Sammy42779** [comment]
Hi, thank you for the great work on this PR!

I’m wondering if there’s any update on PowerPoint .pptx file parsing support?
I’m currently using SimpleDirectoryReader to load .pptx files, but it still seems to raise errors. From the code and discussion here, it looks like the new feature hasn't been fully merged or released yet. Could you please confirm the current status, or suggest a workaround?

Thanks in advance!

**Comment by Jash271** [comment]
> Hi, thank you for the great work on this PR!
> 
> I’m wondering if there’s any update on PowerPoint .pptx file parsing support? I’m currently using SimpleDirectoryReader to load .pptx files, but it still seems to raise errors. From the code and discussion here, it looks like the new feature hasn't been fully merged or released yet. Could you please confirm the current status, or suggest a workaround?
> 
> Thanks in advance!



Hey the MR hasn’t been merged yet! I would recommend overriding the default class and removing the piece where it’s getting the caption for the image 

If you want better text extraction - i.e being able to extract content from tables, charts, etc 
Then I would recommend copy pasting the code for the ppt reader that I have in the MR and give it a custom class name , initialize that and pass it in your file extractor for the simple directory reader

**Comment by Jash271** [comment]
Hi @Sammy42779 @Miarije 

Hope you'll have found it easier to parse ppt's since the PR release 
Wanted to gather some thoughts on it it's been working fine for you'll ?
